cargo-concordium/src/build.rs

```diff
@@ -175,13 +175,26 @@ pub fn build_contract(
     let data_size = (output_bytes.len() - 8) as u32;
     (&mut output_bytes[4..8]).copy_from_slice(&data_size.to_be_bytes());
 
-    let out_filename = out.unwrap_or_else(|| {
-        let extension = match version {
-            utils::WasmVersion::V0 => "v0",
-            utils::WasmVersion::V1 => "v1",
-        };
-        PathBuf::from(format!("{}.{}", filename, extension))
-    });
+    let out_filename = match out {
+        Some(out) => {
+            // A path name and a filename needs to be provided when using the `--out` flag.
+            if out.file_name().is_none() {
```

Comment:
Are you sure that this does the right thing? Looking at the documentation https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.file_name

it looks like if the user supplies `/home/user/Desktop` this check will fail, but the tool will still fail to write the schema out properly.

We probably need to additionally test if the user supplied a directory. https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.is_dir

Reply:
Perhaps I misunderstood the intent though.

Reply:
I added the following to the check:
```
   if out.file_name().is_none() || out.is_dir() {
    // throw error that path + file name has to be given
   }
```